rfd/0080/README.md

```diff
@@ -3,4 +3,54 @@ authors: Casey Bisson <casey.bisson@joyent.com>, Jason Pincin <jason.pincin@joye
 state: predraft
 ---
 
-# RFD 80 projectsConvergence implementation
\ No newline at end of file
+# RFD 80 projectsConvergence implementation
+
+The Mariposa (RFD36) convergence service will be responsible for insuring service goal state is met. It accomplishes this by polling the service and project APIs, the VM API, and by monitoring Changefeed. When a divergence between goal state and actual state is detected, the convergence service will interact with VM API to resolve the discrepancy. 
+
+Note: Direct interaction here means that limits and other controls set for CloudAPI will not be applied to this service. Potential inconsistencies with this approach should be given consideration.
+
+## Goal state
+
+Goal state is obtained by reading the Mariposa Project service API. Changes to goal state will be detected by the Convergence service via Changefeed (assuming the service API can leverage Changefeed for publishing changes), and by polling the Mariposa service API. Although propagation of changes via Changefeed will be faster, polling remains important because by it's nature, Changefeed is not 100% reliable. 
+
+## Actual state
+
+The Convergence service will maintain data structures that represent the actual state. VMAPI Changefeed events will be monitored for changed in the VMs that the Convergence API is responsible for. Just as with goal state, the VM API will also be polled to protect against missed Changefeed events. The Convergence service will ignore events for VMs it did not create. 
+
+## State convergence
+
+Any time there is a divergence between goal state and actual state, the Convergence service has work to do in order to bring the two states back into alignment. At a high level, the paths are:
```

Comment:
Because state changes are not instantaneous, it is desirable to be able to see the queue of convergence tasks that may be executing on a project at any time. In [RFD36/queue](https://github.com/misterbisson/rfd/blob/master/rfd/0036/queue.md) I proposed commands to list the contents of the queue and stop tasks in the queue.

It seems the convergence service is the likely home for any such queue.